tests/flow-pkt-recursion/flow-pkt-recursion-middleware-excluded/test.yaml

```diff
@@ -0,0 +1,18 @@
+# Exclude recursion-level from flow matching
+# Re-use the pcap from the middleware-included test.
+command: |
+  ${SRCDIR}/src/suricata -c "${SRCDIR}/suricata.yaml" -l "${OUTPUT_DIR}" \
+   -r "${TEST_DIR}/../flow-pkt-recursion-middleware-included/test.pcap" \
+   --set "threshold-file=${SRCDIR}/threshold.config" \
+   --set "classification-file=${SRCDIR}`[ -f ${SRCDIR}/etc/classification.config ] && printf '/etc'`/classification.config" \
+   --set "reference-config-file=${SRCDIR}`[ -f ${SRCDIR}/etc/reference.config ] && printf '/etc'`/reference.config" \
+   --set 'decoder.recursion-level.use-for-tracking=false'
+
+checks:
+  - filter:
+      count: 1
+      match:
+        event_type: flow
+        proto: ICMP
+        flow.pkts_toserver: 1
```

Comment:
There does not seem to be a difference with flow-pkt-recursion-middleware-included/test.yaml

Is there ?

Reply:
Correct, there is no difference in the expected flow matching result when the Suricata device is not a tunnel terminator. This is what flow-pkt-recursion-middleware-excluded and flow-pkt-recursion-middleware-included are showing, that for flows on the same recursion level there if no difference with this new config option decoder.recursion-level.use-for-tracking disabled (it is enabled by default).

Reply:
Ok. Is this no difference in behavior not proved by existing tests already ?..

Reply:
Thanks for your comments. The existing test would need to be using tunneled unencrypted packets and explicitly check flow matching to already be proved. A test that matches this behavior might exist, but I thought it was specific enough traffic to add in a test for this feature. Happy to remove if you don't agree.

Reply:
Thanks for the explanation, let's keep it :-)